stl/inc/atomic

```diff
@@ -354,8 +369,18 @@ struct _Atomic_storage {
         _Check_memory_order(_Order);
         const auto _Storage_ptr  = _STD addressof(_Storage);
         const auto _Expected_ptr = _STD addressof(_Expected);
+#if _CMPXCHG_MASK_OUT_PADDING_BITS
+        if constexpr (_Cmpxchg_has_padding_bits_v<_Ty>) {
+            __builtin_zero_non_value_bits(_Expected_ptr);
+        }
+#endif // _CMPXCHG_MASK_OUT_PADDING_BITS

```

Comment:
```suggestion
#if _CMPXCHG_MASK_OUT_PADDING_BITS
        __builtin_zero_non_value_bits(_Expected_ptr);
#endif // _CMPXCHG_MASK_OUT_PADDING_BITS

```

This one doesn't need to be guarded because `__builtin_zero_non_value_bits` will already be a no-op in such cases. We only need the guards if you're going to be doing additional loops or similar with the results.

Ditto the below case if you don't think the 'dirty cache line avoidance' form is worthwhile.

Reply:
Fixed this one, the below apply to local copy, so still needs guard.